broker/README.md

```diff
@@ -0,0 +1,287 @@
+gcloud compute instances start "$instancename" --zone "$zone"
+# night-conductor will get the testid by parsing its own instance name
+# night-conductor shuts down automatically when broker startup is complete
+
+#--- The broker will start ingesting the ZTF stream
+#    and storing/processing the data.
+
+#--- End the night
+# set the attributes
+NIGHT=END
+gcloud compute instances add-metadata "$instancename" --zone="$zone" \
+      --metadata NIGHT="$NIGHT"
+
+# start the VM to trigger the startup script
+gcloud compute instances start "$instancename" --zone "$zone"
+# night-conductor will get the testid by parsing its own instance name
+# the night-conductor VM shuts down automatically when broker shutdown is complete
+```
+
+__Start Night Details:__
+
+Currently, `night-conductor` executes the following to start the night:
+1. Clears the messages from Pub/Sub subscriptions that we use to count the number of elements received and processed each night.
+2. Deploys the two Dataflow jobs and waits for their status to change to "Running".
+3. Starts the [`ztf-consumer`](https://console.cloud.google.com/compute/instancesMonitoringDetail/zones/us-central1-a/instances/ztf-consumer?project=ardent-cycling-243415&tab=monitoring&duration=PT1H&pageState=%28%22duration%22:%28%22groupValue%22:%22P7D%22,%22customValue%22:null%29%29) VM, which is configured with a startup script to connect to ZTF and begin ingesting.
+
+Cloud Functions are always "on"; `night-conductor` does not manage them.
+
+Based on my experience, _what to expect if you allow the broker to ingest a ZTF topic that already contains 200,000 - 400,000 alerts (a typical, active night) as fast as it can_:
+- The consumer should ingest and publish all of the alerts to the `ztf_alert_data` Pub/Sub topic within about 30 minutes.
+- The Avro->GCS storage component (Cloud Function) should process all of the alerts within about 60 minutes.
+- The BigQuery storage component (Dataflow job) will likely get bogged down and take >10 hours to process all of the alerts. I recommend just cancelling the job. This component performs reasonably at the live-stream rate, but it tends to struggle when flooded with alerts. In general, it is the component with the largest lag times and number of dropped alerts. It could use some TLC, or a complete reconfiguring.
+- The value-added processing component (Dataflow job) should process all of the alerts within a few hours.
+
+Note on triggering `night-conductor` to start the night:
+The consumer's connection to ZTF will fail if there is not as least 1 alert in the topic, but the terminal/shell is not released,
+so we can't simply keep trying until the connection succeeds.
+Therefore I am still manually triggering `night-conductor` to start the night after ZTF issues its first alert.
+There _should_ be a programatic way to check whether a topic is available, but I haven't been able to find it yet.
+I have a new lead, so I'll work on it more.
+
```

Comment:
Where should I now look to see what's working and not working?  Where is the dashboard for my test instance?  Where should I expect to see failure messages being sent to?

Reply:
How do a I set up a dashboard that's linked to my, e.g., `ztf-consumer-mwvtest`?

Reply:
I see/agree that it's generally unclear what the user should look at to "see" their testing system running. The best I've been able to do so far is to link you to the production resources to get you in roughly the right place to find your testing resources.

I'm guessing/hoping that there is a way to programmatically create a dashboard using an existing dashboard as a template. I will look and implement a solution if I can find one. I see that it would be incredibly useful.

Reply:
I will also add some links in the READMEs to help the user find the logs.

Reply:
I tried to create a copy of the production Dashboard a modify it to point to `ztf-consumer-mwvtest` but was unable to figure out how to edit to rename the thing it was pointing to.

Reply:
Ya, I think it's not going to be that simple. Every component needs to be renamed, but the character that separates the testid tag from the resource name isn't quite consistent because of differences in restrictions on which characters are allowed in the names of different resources (specifically, `-` vs `_`). For example, the PS topic is `ztf_alert_data-testid`, but the BQ dataset is `ztf_alerts_testid`. I have used `-` as the default, if it is allowed for that resource.

I am working on the consumer simulator right now.* The very next thing I'll do is try to programatically create the dashboard. Doing it manually is way too cumbersome to be reasonable for this task. 

*ZTF does not have any alerts in their system right now, so you can't get any alerts into your test system and all you'd see on the dashboard is a couple of VMs running.